tests/io/v0_0_0/test_caching_backend.py

```diff
@@ -0,0 +1,156 @@
+import contextlib
+import hashlib
+import json
+import os
+import socket
+import subprocess
+import sys
+import time
+import unittest
+
+import numpy as np
+import requests
+import skimage.io
+
+import slicedimage
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from tests.utils import TemporaryDirectory
+
+
+class TestCachingBackend(unittest.TestCase):
```

Comment:
this is awesome. i don't understand a lot of the code, but if this test passes, then i'm extremely happy.